pkg/kv/kvserver/logstore/logstore.go

```diff
@@ -543,6 +545,16 @@ func LoadEntries(
 
 	ents, cachedSize, hitIndex, exceededMaxBytes := eCache.Scan(ents, rangeID, lo, hi, maxBytes)
 
+	budgetFor := func(size uint64) bool {
+		return account == nil || account.Grow(ctx, int64(size)) == nil
```

Comment:
`account.Grow` does not block / wait for the resource, instead it returns an error immediately. So we'll return fewer entries from `Ready` in this case. In the worst case, a range can starve if every its Ready invocation gets no entries because of reaching the memory limit.

One workaround is to always return at least one entry. But this can lead to OOMs.